File: binarySearch.py
def binarySearch(nums,target):
	l,r=0,len(nums)-1
	while l <= r:
	    mid=(r-l)+l//2
	    if nums[mid]==target:
	        return mid
	    elif nums[mid]<target:
	        l=mid+1
	    else:
	        r=mid-1
	return -1

print(binarySearch([-1,0,3,5,9,12],12))

File: binarySearch.java
class Solution{
	public int binarySearch(int[] nums, int target){
		int left=0;
		int right=nums.length-1;
		while (left<=right){
			int mid=(right-left)+left/2; 
			if (nums[mid]==target){
				return mid;
			}else if(nums[mid]<target){
				left=mid+1;
			}
			else{
				right=mid-1;
			}
		}
		return -1;
	}
}

class binarySearch{
	public static void main(String[] args){
		int [] nums={-1,0,3,5,9,12};
		int target= 3;
		Solution s= new Solution();
		System.out.println(s.binarySearch(nums,target));
	}
}